.github/workflows/build-all-and-deploy-devnet.yml

```diff
@@ -255,21 +255,29 @@ jobs:
         with:
           clean: false
           fetch-depth: 0
+      - uses: cachix/install-nix-action@v17
+        with:
+          extra_nix_config: |
+            sandbox = relaxed
+            narinfo-cache-negative-ttl = 0
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
+        with:
+          skipPush: true
+          installCommand: |
+            nix-channel --add https://nixos.org/channels/nixpkgs-22.05-darwin nixpkgs
+            nix-channel --update
+            nix-env -iA nixpkgs.cachix
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          name: ${{  env.NIX_CACHIX }}
 
-      #TODO: use nix tool and just oneliner to dump credentials, no need tasks
-      #TODO: and this task is deprecated
       - uses: google-github-actions/setup-gcloud@v0
         with:
           service_account_key: ${{ secrets.GCP_CREDENTIALS }}
           export_default_credentials: true
 
       - name: Build & Push
         run: |
-          chmod +x ./nix-install-vscode-composable.sh
-          ./nix-install-vscode-composable.sh ${{ env.NIX_INSTALL_SCRIPT }} ${{ env.NIX_NIXPKGS_CHANNEL }} ${{  env.NIX_CACHIX }}
-          source ~/.nix-profile/etc/profile.d/nix.sh && export PATH="/home/$(whoami)/.nix-profile/bin:$PATH"
-          cachix authtoken ${{ secrets.CACHIX_AUTH_TOKEN }}
```

Comment:
no sure. 
- afaik nixos ssh hanged, was it fixed?  just if it will hang - hope not. that why not nixos was  used as base image
- for cacix setup, not sure why you are making bash xplat one liners into github yamls? it is longer, less clear and less reproducible locally.  really github action just add just add layers on top of simple things.

Reply:
got google that is also true.

likely google login is jsut `gsutils login --export ${{ secrets.GCP_CREDENTIALS }}` and that it, better then job as it is more reproducible locally and better documented and xplat accross any ci, including those which nix native or arm enabled:)